packages/edit-post/src/components/options-modal/options.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * WordPress dependencies
+ */
+import { CheckboxControl } from '@wordpress/components';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { withSelect, withDispatch } from '@wordpress/data';
+
+function Option( { label, isChecked, onChange } ) {
+	return (
+		<CheckboxControl
+			className="edit-post-options__option"
+			label={ label }
+			checked={ isChecked }
+			onChange={ onChange }
+		/>
+	);
+}
+
+class DeferredOption extends Component {
+	constructor( { isChecked } ) {
+		super( ...arguments );
+		this.state = {
+			isChecked,
+		};
+	}
+
+	componentWillUnmount() {
+		if ( this.state.isChecked !== this.props.isChecked ) {
+			this.props.onChange( this.state.isChecked );
+		}
+	}
+
+	render() {
+		return (
+			<Option
+				label={ this.props.label }
+				isChecked={ this.state.isChecked }
+				onChange={ ( isChecked ) => this.setState( { isChecked } ) }
+			/>
+		);
+	}
+}
+
+export const EnablePublishSidebarOption = compose(
+	withSelect( ( select ) => ( {
+		isChecked: select( 'core/editor' ).isPublishSidebarEnabled(),
+	} ) ),
+	withDispatch( ( dispatch ) => {
+		const { enablePublishSidebar, disablePublishSidebar } = dispatch( 'core/editor' );
+		return {
+			onChange: ( isEnabled ) => ( isEnabled ? enablePublishSidebar() : disablePublishSidebar() ),
+		};
+	} )
+)( Option );
+
+export const EnableTipsOption = compose(
+	withSelect( ( select ) => ( {
+		isChecked: select( 'core/nux' ).areTipsEnabled(),
+	} ) ),
+	withDispatch( ( dispatch ) => {
+		const { enableTips, disableTips } = dispatch( 'core/nux' );
+		return {
+			onChange: ( isEnabled ) => ( isEnabled ? enableTips() : disableTips() ),
+		};
+	} )
+)( DeferredOption );
```

Comment:
> We should document why we use a deferred option in here. I assume it's to ensure you don't see tips them when interacting with the modal.

That's correct. I've added an explanatory comment in 9ebb7bacd.

> It still raises the question why this needs to be a special case.

I don't want to compromise on what I think is the best UX which is that:

- Deselecting _Enable Pre-publish Checks_ immediately updates the _Publish..._ button to be _Publish_
- Toggling any of the _Document Panels_ options immediately updates the sidebar UI
- Selecting _Enable Tips_ does not display the NUX modal immediately. This is jarring since it interrupts the user's current task

Reply:
This is helpful. It might prevent someone from changing it accidentally to immediate change.